stress/src/async_throughput.rs

```diff
@@ -0,0 +1,295 @@
+        let successful_ops = Arc::new(AtomicU64::new(0));
+
+        // Clone for reporting thread
+        let completed_clone = Arc::clone(&completed_ops);
+        let successful_clone = Arc::clone(&successful_ops);
+        let start_time = Instant::now();
+        let report_interval = config.report_interval;
+
+        // Spawn reporting thread
+        let _reporting_handle = std::thread::spawn(move || loop {
+            std::thread::sleep(report_interval);
+            let ops = completed_clone.load(Ordering::Relaxed);
+            let success = successful_clone.load(Ordering::Relaxed);
+            let elapsed = start_time.elapsed();
+            let throughput = ops as f64 / elapsed.as_secs_f64();
+
+            let success_percentage = if ops > 0 {
+                (success as f64 / ops as f64) * 100.0
+            } else {
+                0.0
+            };
+
+            println!(
+                "Progress: {} ops completed ({} successful, {:.1}%) in {:.2}s = {:.2} ops/sec",
+                ops,
+                success,
+                success_percentage,
+                elapsed.as_secs_f64(),
+                throughput
+            );
+        });
+
+        // Create infinite stream of operations
+        if config.use_spawn {
+            // Create infinite stream of operations with spawning
+            let mut operation_stream = stream::iter(0..)
+                .map(|_| tokio::task::spawn(operation_factory()))
+                .buffer_unordered(config.concurrency);
+
+            // Process stream until interrupted
+            while let Some(result) = operation_stream.next().await {
+                completed_ops.fetch_add(1, Ordering::Relaxed);
+                match result {
+                    Ok(Ok(_)) => {
+                        successful_ops.fetch_add(1, Ordering::Relaxed);
+                    }
+                    Ok(Err(e)) => {
+                        eprintln!("Operation failed: {}", e);
+                    }
+                    Err(e) => {
+                        eprintln!("Task join error: {}", e);
+                    }
+                }
+            }
+        } else {
+            // Create infinite stream of operations without spawning
+            let mut operation_stream = stream::iter(0..)
+                .map(|_| operation_factory())
+                .buffer_unordered(config.concurrency);
+
+            // Process stream until interrupted
+            while let Some(result) = operation_stream.next().await {
+                completed_ops.fetch_add(1, Ordering::Relaxed);
+                match result {
+                    Ok(_) => {
+                        successful_ops.fetch_add(1, Ordering::Relaxed);
+                    }
+                    Err(e) => {
+                        eprintln!("Operation failed: {}", e);
+                    }
+                }
+            }
+        }
+
+        // Return final stats
+        let duration = start_time.elapsed();
```

Comment:
nit: Since this function would never return and relies on the user to Ctrl +C, it doesn't really need a return value.